### PR TITLE
migration to go modules, removed old CRDs, re-vendored packages

### LINKED ISSUE #15: Go modules
GO modules are now the standard way of dependency management. This project was boostraped with an older version of operator-sdk which used `dep` for dependency managemen by default. Dependency management needs to be refactored to use go modules.

### PR DESCRIPTION
**Pull request by hitman99**
This is a massive change due to the fact that I re-populated the `vendor` dir afer migration from dep to go modules using `go mod vendor`. Closes #15

### CODE REVIEW

**Comment by arnarg** [comment]
This is looking good @hitman99. I just had one question in my review.

**Comment by hitman99** [comment]
@arnarg I cannot find your comments on this PR, it still says a review is pending

**Comment by arnarg** [comment]
Interesting, can you see it if you look at files changed?

**Comment by arnarg** [comment]
oh looks like inline comments are not visible until I post a full review

File: pkg/apis/db/v1alpha1/zz_generated.openapi.go [hunk 1]
```diff
@@ -11,13 +11,13 @@ import (
...
 func GetOpenAPIDefinitions(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
 	return map[string]common.OpenAPIDefinition{
-		"github.com/movetokube/postgres-operator/pkg/apis/db/v1alpha1.Postgres":           schema_pkg_apis_db_v1alpha1_Postgres(ref),
-		"github.com/movetokube/postgres-operator/pkg/apis/db/v1alpha1.PostgresRoles":      schema_pkg_apis_db_v1alpha1_PostgresRoles(ref),
-		"github.com/movetokube/postgres-operator/pkg/apis/db/v1alpha1.PostgresSpec":       schema_pkg_apis_db_v1alpha1_PostgresSpec(ref),
-		"github.com/movetokube/postgres-operator/pkg/apis/db/v1alpha1.PostgresStatus":     schema_pkg_apis_db_v1alpha1_PostgresStatus(ref),
-		"github.com/movetokube/postgres-operator/pkg/apis/db/v1alpha1.PostgresUser":       schema_pkg_apis_db_v1alpha1_PostgresUser(ref),
-		"github.com/movetokube/postgres-operator/pkg/apis/db/v1alpha1.PostgresUserSpec":   schema_pkg_apis_db_v1alpha1_PostgresUserSpec(ref),
-		"github.com/movetokube/postgres-operator/pkg/apis/db/v1alpha1.PostgresUserStatus": schema_pkg_apis_db_v1alpha1_PostgresUserStatus(ref),
+		"./pkg/apis/db/v1alpha1.Postgres":           schema_pkg_apis_db_v1alpha1_Postgres(ref),
```
[the remark this answers is not in the document]
Comment (hitman99) [reply pkg/apis/db/v1alpha1/zz_generated.openapi.go]: Yeah, it looked a bit strange to me too, but it does compile though. I used operator-sdk v0.13.0 to re-generate the code. At some point we might want to update the dependencies from operator-framework 0.7 to 0.13

**Pull request merged**